### PR TITLE
feat/split: Split network page into network and search

### LINKED ISSUE #53: Split network page
Network should have two new pages: search (footer) and "my network" (sidebar)
- [x] Split pages
- [x] Make code reusable across pages 

Search:
- [x] Show recommended users by default
- [x] Functionality for connection request
- [x] Show data from backend, not dummy data
- [x] Search for any user
- [ ] ~~For contacts, functionality for inviting to activity~~ 

Network:
- [x] Show my connections
- [x] Show new requests, and accepted requests
- [x] Show real data
- [ ] ~~Show invite to activity functionality~~

EDIT:
Activity functionality will be addressed in [https://github.com/askbk/inmarket-frontend/pull/84](https://github.com/askbk/inmarket-frontend/pull/84)

### PR DESCRIPTION
**Pull request by edvardvb**
Splits network and search pages, while reusing code for displaying contacts and potential contacts. Lots of work to do still, especially related to connecting this to the backend, but the split and display is coming along nicely.

- Depends on:
  - #77 (MERGED)
  - https://github.com/askbk/inmarket-backend/pull/19 (MERGED)
  - https://github.com/askbk/inmarket-backend/pull/21

Based on #53, but does not currently complete all subtasks.

### Search page
**Current status:**
- Presents recommended users initially
- Searches real users
- Functionality for requesting, accepting, and denying contact.
  - State updates automatically, so that the user is presented properly in the list.

**Nice to have:** 
- _BACKEND_ Search needs improvement. Currently only searches for exact strings in `firstName + lastName`
  - See https://github.com/askbk/inmarket-backend/pull/22

### Network page
**Current status:** 
- Presents separate lists for pending requests and existing contacts.
- Existing contacts are searchable
- Functionality for accepting and denying contacts.
  - State updates automatically, moving the contact to list of existing contacts, or removing them when denying.

**Depends on**
- _BACKEND_ Need endpoints for real data.
  - This PR relies on the routes added in https://github.com/askbk/inmarket-backend/pull/21, and currently assumes that these will work 

### General
**To do:**
- Functionality for activity invites
  - See #52 and https://github.com/askbk/inmarket-frontend/tree/feat/activities
- _BACKEND_ Pass connection status when fetching users. The lists need to know whether a user is connected, has requested contact etc. to display them properly.
  - See https://github.com/askbk/inmarket-backend/issues/14
  - Currently displays random placeholder connection statuses
- _BACKEND_ Pass user ratings when fetching users.
  - See https://github.com/askbk/inmarket-backend/issues/14
  - Currently displays random placeholder ratings

### CODE REVIEW

**Comment by edvardvb** [comment]
### Current status:
**Search page:**
image

**Network page:**
image

### Remaining work:

- Ensure changes are pushed to state tree when contacting, accepting, denying or inviting a user, so that their list element updates accordingly.
  - Difficult when relying on test data. Might be easier when we can fetch real data.
- Implement invite functionality. How do we select which activity to invite the user to upon clicking "invite"?
- Fetch real data. Needs backend work.
- Implement real search for all users in DB. Needs backend work.
- (Implement "se logg"?)

Could consider merging as is, and continuing in a new PR.

**Comment by askbk** [comment]
Looking good!

- For now, the "Invite" button should let the current user send an activity invitation using the activity creation form that is currently being worked on in feat/activities by @filipot .
- "Se logg" can simply be hidden for now, I think.

The way contacts are modeled in the database is by having a Contact(contacter, contactee) table. This means that if a user A has sent a contact request to user B, user B has to send a "contact request" to user A in order to accept.

Basically, you should be able to use the same API endpoint for both sending and accepting contact requests. Declining will be handled by a different enpoint, but has yet to be implemented.

**Comment by edvardvb** [comment]
### Search page
**Current status:**
- Presents recommended users initially
- Searches real users (depending on https://github.com/askbk/inmarket-backend/pull/19)
- Functionality for requesting and accepting contact.
  - State updates automatically, so that the user is presented properly in the list.

**To do:**
- Functionality for invites and denying contact requests
- _BACKEND_ Search needs improvement. Currently only searches for exact strings in `firstName + lastName`
  - See https://github.com/askbk/inmarket-backend/pull/22

### Network page
**Current status:** 
- Presents separate lists for pending requests and existing contacts.
- Existing users are searchable
- Functionality for accepting contacts.
  - State updates automatically, moving the contact to list of existing contacts.

**To do:**
- Functionality for invites
- _BACKEND_ Need endpoints for real data.
  - One endpoint for pending requests
  - One endpoint for current contacts

**Comment by edvardvb** [comment]
Original description has been updated with PR status.

This is what's left to do until this is completely done.

1. _BACKEND_ Network needs endpoints for real data.
     - This PR relies on the routes added in askbk/inmarket-backend#21, and currently assumes that these will work. Therefore, that PR must be completed before this.
2. Functionality for activity invites
    - See #52 and https://github.com/askbk/inmarket-frontend/tree/feat/activities
    - Currently, invite button does nothing
3. _BACKEND_ Pass connection status when fetching users. The lists need to know whether a user is connected, has requested contact etc. to display them properly.
    - See askbk/inmarket-backend#14
    - Currently displays random placeholder connection statuses
4. _BACKEND_ Pass user ratings when fetching users.
    - See https://github.com/askbk/inmarket-backend/issues/14
    - Currently displays random placeholder ratings

I suggest waiting until 1. is complete, and then removing placeholder data before merging this PR. One could also argue that it would be best to wait until 3. is completed as well. The last points, 2. and 4., are not strictly relevant right now.

I also suggest moving these subtasks to a new issue, and to close #53 upon merge.

**Review by askbk — APPROVED** [review]
